src/toil/jobStores/aws/jobStore.py

```diff
@@ -1023,27 +1078,24 @@ def strOrNone(s):
                 return s if s is None else str(s)
 
             # ownerID and encrypted are the only mandatory attributes
-            ownerID = strOrNone(item.get('ownerID'))
-            encrypted = item.get('encrypted')
+            ownerID, encrypted, version, checksum = SDBHelper.get_attributes_from_item(item, ["ownerID", "encrypted", "version", "checksum"])
             if ownerID is None:
                 assert encrypted is None
                 return None
             else:
-                version = strOrNone(item['version'])
-                checksum = strOrNone(item.get('checksum'))
                 encrypted = strict_bool(encrypted)
-                content, numContentChunks = cls.attributesToBinary(item)
+                content, numContentChunks = cls.attributesToBinary(item["Attributes"])
                 if encrypted:
                     sseKeyPath = cls.outer.sseKeyPath
                     if sseKeyPath is None:
                         raise AssertionError('Content is encrypted but no key was provided.')
                     if content is not None:
                         content = encryption.decrypt(content, sseKeyPath)
-                self = cls(fileID=item.name, ownerID=ownerID, encrypted=encrypted, version=version,
+                self = cls(fileID=item["Name"], ownerID=ownerID, encrypted=encrypted, version=version,
                            content=content, numContentChunks=numContentChunks, checksum=checksum)
                 return self
 
-        def toItem(self):
+        def toItem(self) -> Tuple[Dict[str, str], int]:
             """
             Convert this instance to an attribute dictionary suitable for SDB put_attributes().
```

Comment:
This might not be the right way to describe the result anymore if it now requires further processing with `SDBHelper` to actually send to AWS. Maybe we want to call it just a dictionary from attribute name to value?